modules/Block/Block.resolvers.js

```
@@ -0,0 +1,369 @@
+    const existingBlock = await BlockModel.exists({
+      _id: id,
+      block_status: "ACTIVE",
+    });
+    // *************** If block is not found or already completed, throw an error
+    if (!existingBlock) {
+      throw new ApolloError(
+        "Block not found or already completed",
+        "NOT_FOUND"
+      );
+    }
+
+    // *************** Soft delete: update block_status and set deleted_at timestamp
+    const updateResult = await BlockModel.updateOne(
+      { _id: id },
+      {
+        block_status: "COMPLETED",
+        deleted_at: new Date(),
+      }
+    );
+    return updateResult;
+    // *************** If an error occurs during the update, throw an ApolloError with details
+  } catch (error) {
+    throw new ApolloError("Failed to delete block", "BLOCK_DELETION_FAILED", {
+      error: error.message,
+    });
+  }
+}
+
+// *************** LOADER ***************
+
+/**
+ * subject - Resolver to load multiple Subject documents based on subject_ids in the parent Block object.
+ *
+ * @param {Object} parent - The parent object, typically a Block, which contains subject_ids.
+ * @param {Object} _ - Unused GraphQL args parameter.
+ * @param {Object} context - The GraphQL context object containing the SubjectLoader instance.
+ * @param {Object} context.dataLoaders.SubjectLoader - DataLoader for batching and caching Subject fetches.
+ * @returns {Promise<Array>} A promise that resolves to an array of Subject documents, or an empty array if none.
+ */
+async function subject(parent, _, context) {
+  if (parent.subject_ids) {
+    // *************** Use the Subject Loader to load many subject documents by its ID
+    const toSubjectList = await context.dataLoaders.SubjectLoader.loadMany(
+      parent.subject_ids
+    );
+    // *************** Return the loaded subject documents
+    return toSubjectList;
+  } else {
+    // *************** If no subject_id is present in the parent object, return empty
+    return [];
```

fail fast, if not exists, directly return array empoty

---

I've updated it at commit ID: f12f939ef5066b0891c786cd173109df348e77be